security/pkg/pki/ca/ca.go

```diff
@@ -82,7 +83,8 @@ type IstioCA struct {
 
 	keyCertBundle util.KeyCertBundle
 
-	livenessProbe *probe.Probe
+	monitoringCounters monitoring.CAOperationCounters
```

Comment:
We should make `pkg/pki` not depend on monitoring libraries. Instead can we:
- Define the customized error code for signing related logic, say TTL_FAIL, INVALID_PEM, etc.
- ca.go returned the customized error to server.go
- The metrics defined there can do signingError[errorCode].Inc()

In this way, we also don't have to define one monitoring metric per error type, which makes code simpler, also in monitoring post-processing side, we can use more expressive to query the motoring metrics.